cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/PreventBrokerScaleDownUtils.java

```diff
@@ -0,0 +1,195 @@
+     * @param kafkaStatus          The Kafka Status class for adding conditions to it during the reconciliation
+     * @param secretOperator       Secret operator for working with Secrets
+     * @param adminClientProvider  Used to create the Admin client instance
+     * @param currentReplicas      The current number of replicas
+     * @return  returns a boolean future based on the outcome of the check
+     */
+    public static Future<Boolean> canScaleDownBrokerCheck(Vertx vertx, Reconciliation reconciliation, KafkaCluster kafka, KafkaStatus kafkaStatus,
+                                                      SecretOperator secretOperator, AdminClientProvider adminClientProvider, int currentReplicas) {
+
+        Promise<Boolean> cannotScaleDown = Promise.promise();
+        LOGGER.infoCr(reconciliation, "Entering canScaleDownBrokers");
+        ReconcilerUtils.clientSecrets(reconciliation, secretOperator)
+                .compose(compositeFuture -> {
+                    Promise<Void> resultPromise = Promise.promise();
+                    vertx.createSharedWorkerExecutor("kubernetes-ops-pool").executeBlocking(
+                            future -> {
+                                Admin kafkaAdmin = null;
+                                final Future<Collection<TopicDescription>> descriptions;
+                                try {
+                                    String bootstrapHostname = KafkaResources.bootstrapServiceName(reconciliation.name()) + "." + reconciliation.namespace() + ".svc:" + KafkaCluster.REPLICATION_PORT;
+                                    LOGGER.infoCr(reconciliation, "Creating AdminClient for cluster {}/{}", reconciliation.namespace(), kafka.getName());
+                                    kafkaAdmin = adminClientProvider.createAdminClient(bootstrapHostname, compositeFuture.resultAt(0), compositeFuture.resultAt(1), "cluster-operator");
+
+                                    Admin finalKafkaAdmin = kafkaAdmin;
+                                    Future<Set<String>> topicNames = topicNames(reconciliation, finalKafkaAdmin);
+
+                                    descriptions = topicNames.compose(names -> {
+                                        LOGGER.infoCr(reconciliation, "Topic names {}", names);
+                                        return describeTopics(reconciliation, finalKafkaAdmin, names);
+                                    });
+                                    descriptions
+                                            .compose(topicDescriptions -> {
+                                                LOGGER.infoCr(reconciliation, "Got {} topic descriptions", topicDescriptions.size());
+
+                                                for (int i = 0; i < currentReplicas - kafka.getReplicas(); i++) {
+
+                                                    if (brokerHasAnyReplicas(reconciliation, topicDescriptions, currentReplicas - i - 1)) {
```

Comment:
Would it be better if we stored the result of `brokerHasAnyReplicas` in a boolean variable, then used it for the conditional and `cannotScaleDown.complete`, so that `brokerHasAnyReplicas` is only called once per iteration?